db/anonymise/rules.rb

```diff
@@ -0,0 +1,144 @@
+require 'faker'
+
+NINO_REGEXP = /^[A-CEGHJ-PR-TW-Z]{1}[A-CEGHJ-NPR-TW-Z]{1}[0-9]{6}[A-DFM]{1}$/.freeze
+
+{
+  # data is set by us and should not leak data
+  # active_storage_attachments,
+  # active_storage_blobs
+  # active_storage_variant_records
+  # actor_permissions
+  # application_proceeding_types
+  # application_proceeding_types_scope_limitations
+  # application_scope_limitations
+  # ar_internal_metadata
+  # attachments
+  # bank_holidays
+  # benefit_check_results
+  # ccms_submission_documents
+  # ccms_submissions
+  # cfe_results
+  # cfe_submission_histories
+  # cfe_submissions
+  # irregular_incomes
+  # legal_aid_application_transaction_types
+  # merits_assessments
+  # offices
+  # offices_providers
+  # other_assets_declarations
+  # permissions
+  # policy_disregards
+  # proceeding_type_scope_limitations
+  # proceeding_types
+  # savings_amounts
+  # scope_limitations
+  # secure_data
+  # service_levels
+  # settings
+  # state_machine_proxies
+  # transaction_types
+  # true_layer_banks
+  # vehicles
+
+  addresses: {
+    address_line_one: -> { Faker::Address.street_address },
+    address_line_two: nil,
+    city: -> { Faker::Address.city },
+    county: -> { Faker::Address.city },
+    postcode: -> { Faker::Address.zip },
+    organisation: nil
+  },
+  admin_users: {
+    username: -> { Faker::Internet.username },
+    email: -> { Faker::Internet.email }
+  },
+  applicants: {
+    first_name: -> { Faker::Name.first_name },
+    last_name: -> { Faker::Name.last_name },
+    email: -> { Faker::Internet.email },
+    date_of_birth: -> { Faker::Date.birthday },
+    national_insurance_number: -> { Faker::Base.regexify(NINO_REGEXP) }
+  },
+  bank_account_holders: {
+    full_name: -> { Faker::Name.name },
+    date_of_birth: -> { Faker::Date.birthday },
+    addresses: nil
+  },
+  bank_accounts: {
+    name: 'Anonymous bank account',
+    account_number: -> { Faker::Bank.account_number },
+    sort_code: -> { Faker::Base.regexify(/^[0-9]{2}-[0-9]{2}-[0-9]{2}$/) }
+  },
+  bank_errors: {
+    bank_name: -> { Faker::Bank.name }
+  },
+  bank_providers: {
+    bank_name: -> { Faker::Bank.name }
+  },
+  bank_transactions: {
+    description: -> { Faker::Company.industry },
+    merchant: -> { Faker::Company.name }
+  },
+  cash_transactions: {
+
+  },
+  ccms_submission_histories: {
+    details: nil,
+    request: nil,
+    response: nil
+  },
+  cfe_results: {
+    result: nil
+  },
+  cfe_submission_histories: {
+    request_payload: nil,
+    response_payload: nil
+  },
+  cfe_submissions: {
+    cfe_result: nil
+  },
+  debugs: {
+
+  },
+  dependants: {
+    name: -> { Faker::Name.name },
+    date_of_birth: -> { Faker::Date.birthday }
```

Comment:
Dont we use the dependant DOB to calculate if they are a child? If so we may want to retain this DOB as is in the original database.

Reply:
👍